### PR TITLE
Validate identifiers by checking conversion to snake case and back

### PR DESCRIPTION
**Pull request by lbam**

### CODE REVIEW

**Comment by lbam** [comment]
Just tested this on the dataset from https://github.com/Amsterdam/amsterdam-schema/pull/339 and it works.

**Review by gkoller — APPROVED** [review]
Yeach, another validator. Nice work! 👍

File: src/schematools/validation.py [hunk 1]
```diff
@@ -127,6 +127,28 @@ def run_all(self) -> Iterator[ValidationError]:
...
+def _validate_camelcase(ident: str) -> Iterator[ValidationError]:
+    if ident == "":
+        return ValidationError("CamelCaseValidator", "empty identifier not allowed")
+    camel = toCamelCase(to_snake_case(ident))
+    if camel == ident:
+        return
+    msg = f"{ident} does not survive conversion to snake case and back, suggestion: {camel}"
+    return ValidationError("CamelCaseValidator", msg)
+
+
```
Comment (gkoller) [inline src/schematools/validation.py:151]: There are some MyPy errors:

```
schematools/validation.py:139: error: Incompatible types in "yield" (actual type "Iterator[ValidationError]", expected type "ValidationError")
schematools/validation.py:144: error: Incompatible return value type (got "ValidationError", expected "Iterator[ValidationError]")
schematools/validation.py:147: error: Return value expected
schematools/validation.py:149: error: Incompatible return value type (got "ValidationError", expected "Iterator[ValidationError]")
```

Comment (gkoller) [reply src/schematools/validation.py:151]: Easily resolved by changing the function return type on line 142 to `Optional[ValidationError]` and the return statement on line 147 to `return None`.

**Review by jjmurre — APPROVED** [review]
LGTM, good idea to catch strange edge cases as early as possible.

**Pull request merged**